nir/src/main/scala/scala/scalanative/nir/Vals.scala

```diff
@@ -26,17 +29,33 @@ sealed abstract class Val {
 }
 object Val {
   // low-level
-  final case object None                                      extends Val
-  final case object True                                      extends Val
-  final case object False                                     extends Val
-  final case class Zero(of: nir.Type)                         extends Val
-  final case class Undef(of: nir.Type)                        extends Val
-  final case class I8(value: Byte)                            extends Val
-  final case class I16(value: Short)                          extends Val
-  final case class I32(value: Int)                            extends Val
-  final case class I64(value: Long)                           extends Val
-  final case class F32(value: Float)                          extends Val
-  final case class F64(value: Double)                         extends Val
+  final case object None               extends Val
+  final case object True               extends Val
+  final case object False              extends Val
+  final case class Zero(of: nir.Type)  extends Val
+  final case class Undef(of: nir.Type) extends Val
+  final case class I8(value: Byte)     extends Val
+  final case class I16(value: Short)   extends Val
+  final case class I32(value: Int)     extends Val
+  final case class I64(value: Long)    extends Val
+  final case class F32(value: Float) extends Val {
+    override def equals(that: Any): Boolean =
+      if (that.isInstanceOf[F32]) {
+        val theseBits = floatToRawIntBits(value)
+        val thoseBits = floatToRawIntBits(that.asInstanceOf[F32].value)
+        theseBits == thoseBits
+      } else
+        false
+  }
+  final case class F64(value: Double) extends Val {
+    override def equals(that: Any): Boolean =
+      if (that.isInstanceOf[F64]) {
+        val theseBits = doubleToRawLongBits(value)
+        val thoseBits = doubleToRawLongBits(that.asInstanceOf[F64].value)
+        theseBits == thoseBits
+      } else
+        false
+  }
```

Comment:
I tried avoiding pattern matching, because this method is likely to be hot, and I wanted to make sure we don't worsen performance by some pattern matching artifacts.

Shall I change it?